### PR TITLE
controlplane/control: Introduce peer manager

### PR DESCRIPTION
**Pull request by orozery**
This PR adds the peerManager which is responsible for monitoring peers, and updating peer status (reachable or not).

### CODE REVIEW

File: pkg/controlplane/control/peer.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	ticker := time.NewTicker(heartbeatInterval)
+	defer ticker.Stop()
+
+	backoffConfig := backoff.NewExponentialBackOff()
+
+	reachable := false
+	reachableCond := metav1.Condition{
+		Type:   v1alpha1.PeerReachable,
+		Status: metav1.ConditionFalse,
+		Reason: "Heartbeat",
```
Comment (kfirtoledo) [inline pkg/controlplane/control/peer.go:86]: Where do you update the time?

File: pkg/controlplane/control/peer.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+				backoffConfig.MaxElapsedTime = 0
+			}
+
+			reachable = heartbeatOK
+
+			m.lock.Lock()
+			meta.SetStatusCondition(&m.pr.Status.Conditions, reachableCond)
+			m.lock.Unlock()
+
+			m.statusCallback(m.pr)
```
Comment (kfirtoledo) [inline pkg/controlplane/control/peer.go:115]: Can you add a comment why you call the callback

File: pkg/controlplane/control/peer.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+func (m *peerManager) Name() string {
+	return "peerManager"
+}
+
+func (m *peerManager) SetStatusCallback(callback func(*v1alpha1.Peer)) {
+	m.statusCallback = callback
+}
+
+// Start the peer manager.
+func (m *peerManager) Start() error {
```
Comment (kfirtoledo) [inline pkg/controlplane/control/peer.go:163]: Where do you call this function?

File: pkg/controlplane/control/peer.go (hunk 2 above)
Comment (orozery) [reply pkg/controlplane/control/peer.go:115]: Sure (it's for non-CRD mode, which does not watch Peer status).

File: pkg/controlplane/control/peer.go (hunk 3 above)
Comment (orozery) [reply pkg/controlplane/control/peer.go:163]: It's called by the runnable manager, which subscribes here:
https://github.com/clusterlink-net/clusterlink/blob/b41c7721d4260568a40aa1f79fc03e26859f90e8/cmd/cl-controlplane/app/server.go#L205

**Review by kfirtoledo — APPROVED** [review]

**Pull request merged**